### PR TITLE
DS_Week3_복습과제_여유진

### PR DESCRIPTION
**Pull request by YEOYUJIN**

### CODE REVIEW

**Comment by kkyung0131** [comment]
1.3
index 열이 잘 빠졌는지 확인하기 위해 .head( )를 사용할 수도 있습니다. 
또한 열의 이름을 알고 싶을 때 
데이터명.columns로 확인할 수 있습니다!(ex. train_df.columns)

1.6 
더 간단하게 인코딩하신 것 같아요!
레이블 인코딩을 할 수 있는 다른 코드도 보시면 좋을 것 같아 적어드립니다!
1)(파머완 코드)
items = ['white','red']
encoder = LabelEncoder()
encoder.fit(items)
labels = encoder.transform(items)
train_df['type'] = encoder.fit_transform(train_df['type'].values)
print(train_df['type'].value_counts())

2)(차수빈벗 코드)
encoder = LabelEncoder() # 라벨 인코더 객체 생성
encoder.fit(train_df['type'])
train_df['type'] = encoder.transform(train_df['type']) # label-encoding을 시킨 결과를 다시 df의 컬럼에 저장
train_df['type'].value_counts()
# red, white -> 0,1로 encoding

3)(조혜빈벗 코드)
items=['type']
le=LabelEncoder()
for i in items:
traindrop[i]=le.fit_transform(traindrop[i])
traindrop['type'].value_counts()  

1.8
정확도를 소수점 4자리까지 뽑고 싶다면
print('결정 트리 예측 정확도: {0:.4f}'.format(accuracy))
으로 작성할 수 있습니다!

2.1
코드의 결괏값이 다른 이유는 문제에 제시된 LogisticRegression의 파라미터 값을 지정 안 해주셔서 그런 것 같아요. 다른 코드들은 문제가 없으니까 LogisticRegression의 solver 파라미터만 추가해주시면 될 것 같아요!!
lr_clf = LogisticRegression(solver="liblinear")
2.2
2번 역시 1번에서 LogistiRegression의 파라미터를 설정해주지 않은데서 비롯한 차이인 것 같습니다! 코드 자체는 답이랑 똑같습니다!!
2.6
wine 데이터의 경우 파마완에서 사용한 데이터와는 다르게 변수의 개수가 몇 개 안되기 때문에 막대 그래프 상위 20개만 보여달라고 할 필요는 없을 것 같아서 해당 필요 없는 부분만 삭제해주시면 좋을 것 같아요.
ftr_top = ftr_importances.sort_values(ascending=False)

3.2
문제에 언급된 부분이 없음에도 cv를 설정해주신 점 좋아요!

3.3
best_estimator_를 이용하는 방법 말고 다른 방법도 있어서 적어드립니다.
lin_svm = SVC(kernel='linear',C=1)
lin_svm.fit(X_train,y_train)
pred = lin_svm.predict(X_test)
accuracy = accuracy_score(y_test, pred)
print('accuracy : {0:.4f}'.format(accuracy))
